src/commands/generate/__tests__/resources/test-cases/example-types/product/bearerAliasExample.ts

```diff
@@ -0,0 +1 @@
+export type IBearerAliasExample = string & { __conjure_type?: "BearerAliasExample" };
```

Comment:
NOTE: should the field value contain the package name as well?

Reply:
I think it should. In which case here the typings would become:

```Typescript
export type IBearerAliasExample = string & { __conjure_type?: "com.palantir.product:BearerAliasExample" };
```

@ferozco  @Que3216  for thoughts

Reply:
We probably want package name here, otherwise aliases with the same name but from different packages will be assignable to each other (and I don't think we want that).

Reply:
agree with @patrickszmucer  the fully qualified classname sounds like the right thing

Reply:
It is a common pattern to share identifiers between multiple services e.g. Object RIDs, Object Type Rids. And I'd imagine it'd be pain having to cast these all the time (which version of ObjectRid from the 10 conjure services would you even use to type the ObjectRids in your store?).

I'd guess that 95% of the time developers would want types of the same name from different services to be compatible, though there will be some clashes. Sometimes those clashes will be zero risk e.g. many services probably have a StateRid alias, but it's pretty unlikely you'll get a bug from accidentally passing the one from the wrong service. Some of them may have some risk, e.g. if Gotham has an ObjectId, you could imagine someone accidentally passing that to Foundry. You _could_ argue that where there are consequential clashes you should rename your types.

If we went ahead with adding the package, would the proposal here be that services put these rids under a common package name, so you just have 'com.palantir.foundry:ObjectRid' rather than 'com.palantir.foundry.monitoring.ObjectRid', 'com.palantir.foundry.oss.ObjectRid' etc.? I guess the first time we make this change to each service it'd technically be an API break for any Conjure Java clients right?

Note: we'd also have to update the copy conjure excavator to avoid shading certain RIDs (https://github.palantir.build/foundry/workshop-server/pull/2345/files)

## Four alternative solutions

### 1. As proposed, add the package straight into the conjure_type

### 2. Add another __conjure_package property

If we want to provide FE's maximum flexibility would could do:

```ts
export type IBearerAliasExample = string & {
   __conjure_package?: "com.palantir.product"
   __conjure_type?: "BearerAliasExample"
};
```

Then if FE's want they can use a helper like:

```ts
export IgnorePackage<T> = Omit<T, "__conjure_package">
```

And if FE's want they can package agnostic versions of these types:

```
export type IObjectRid = { __conjure_type?: "ObjectRid" };
```

And so use that in their internal store/state:

```
{
   selectedObject: IObjectRid
}
``` 

Then no cast is needed if you do an OSS search, get back an OSS.ObjectRid, store it in your state, and then send that RID back out to Phonograph. But if you really do want types from different packages to be treated as different then you can.

### 3. Make it configurable

Allow it to be configured on an alias by alias, or service by service basis.

### 4. Don't include the package at all

Maybe solution 2 is the nicest?

Reply:
@raiju for SA (who's been thinking about other Conjure changes to make cross-service stuff easier)

Reply:
In java code for now RIDs have to be unwrapped/wrapped explicitly right when we are trying to use `ObjectTypeId` from one service in another service? So I guess we'd end up in the same situation where a cast has to be explicitly made in frontend apps when this. happens

I do like 2 better as well, as it provides more flexibility indeed, and allows for keeping a more "neutral" rid in state.

Reply:
If alias'd, yes, they have to be unwrapped/re-wrapped explicitly. Agreed, I think 2 makes a lot of sense